packages/account-tree-controller/src/backup-and-sync/user-storage/network-operations.ts

```diff
@@ -0,0 +1,289 @@
+    const walletData = await context.messenger.call(
+      'UserStorageController:performGetStorage',
+      `${USER_STORAGE_WALLETS_FEATURE_KEY}.${USER_STORAGE_WALLETS_FEATURE_ENTRY_KEY}`,
+      entropySourceId,
+    );
+    if (!walletData) {
+      return null;
+    }
+
+    try {
+      backupAndSyncLogger(
+        `Retrieved wallet data from user storage: ${JSON.stringify(walletData)}`,
+      );
+      return parseWalletFromUserStorageResponse(walletData);
+    } catch (error) {
+      backupAndSyncLogger(
+        `Failed to parse wallet data from user storage: ${error instanceof Error ? error.message : String(error)}`,
+      );
+      return null;
+    }
+  });
+};
+
+/**
+ * Pushes the wallet to user storage.
+ *
+ * @param context - The backup and sync context.
+ * @param wallet - The wallet to push to user storage.
+ * @returns A promise that resolves when the operation is complete.
+ * @throws When network operations fail after maximum retry attempts.
+ * @throws When messenger calls to UserStorageController fail due to authentication errors, encryption/decryption failures, or network issues.
+ * @throws When JSON.stringify fails on the formatted wallet data.
+ */
+export const pushWalletToUserStorage = async (
+  context: BackupAndSyncContext,
+  wallet: AccountWalletEntropyObject,
+): Promise<void> => {
+  return executeWithRetry(async () => {
+    const formattedWallet = formatWalletForUserStorageUsage(context, wallet);
+    const stringifiedWallet = JSON.stringify(formattedWallet);
+    const entropySourceId = wallet.metadata.entropy.id;
+
+    backupAndSyncLogger(`Pushing wallet to user storage: ${stringifiedWallet}`);
+
+    return await context.messenger.call(
+      'UserStorageController:performSetStorage',
+      `${USER_STORAGE_WALLETS_FEATURE_KEY}.${USER_STORAGE_WALLETS_FEATURE_ENTRY_KEY}`,
+      stringifiedWallet,
+      entropySourceId,
+    );
+  });
+};
+
+/**
+ * Retrieves all groups from user storage.
+ *
+ * @param context - The backup and sync context.
+ * @param entropySourceId - The entropy source ID.
+ * @returns An array of groups from user storage.
+ * @throws When network operations fail after maximum retry attempts.
+ * @throws When messenger calls to UserStorageController fail due to authentication errors, encryption/decryption failures, or network issues.
+ */
+export const getAllGroupsFromUserStorage = async (
+  context: BackupAndSyncContext,
+  entropySourceId: string,
+): Promise<UserStorageSyncedWalletGroup[]> => {
+  return executeWithRetry(async () => {
+    const groupData = await context.messenger.call(
+      'UserStorageController:performGetStorageAllFeatureEntries',
+      `${USER_STORAGE_GROUPS_FEATURE_KEY}`,
+      entropySourceId,
+    );
+    if (!groupData) {
+      return [];
+    }
+
+    const allGroups = groupData
+      .map((groupStringifiedJSON) => {
+        try {
+          return parseGroupFromUserStorageResponse(groupStringifiedJSON);
```

Comment:
Very super nit: The "legacy" counterpart (for accounts) of this function uses a different name (despite being super similar), so just in case you want, we could use the same here:

```suggestion
      .map((stringifiedGroup) => {
        try {
          return parseGroupFromUserStorageResponse(stringifiedGroup);
```